packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,29 +190,297 @@ class CupertinoAlertDialog extends StatelessWidget {
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+    final double contentHeight = _content.getMaxIntrinsicHeight(width);
+    final double actionsHeight = _actions.getMaxIntrinsicHeight(width);
+    final double height = max(contentHeight, actionsHeight);
+
+    if (height.isFinite)
+      return height;
+    return 0.0;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    RenderBox _content;
+    RenderBox _actions;
+    getChildrenAsList().forEach((RenderBox child) {
+      CupertinoDialogRenderBoxParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+    final double contentHeight = _content.getMaxIntrinsicHeight(width);
+    final double actionsHeight = _actions.getMaxIntrinsicHeight(width);
+    final double height = min(contentHeight, actionsHeight);
+
+    if (height.isFinite)
+      return height;
+    return 0.0;
+  }
+
+  @override
+  void performLayout() {
+    RenderBox _content;
+    RenderBox _actions;
+    getChildrenAsList().forEach((RenderBox child) {
+      CupertinoDialogRenderBoxParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+    // TODO(mattcarroll): parameterize actions count.
+    final actionsCount = 3;
+
+    double minActionSpace = 0.0;
+    if (actionsCount > 0 && actionsCount <= 2) {
+      minActionSpace = _kButtonHeight;
+    } else {
+      minActionSpace = 1.5 * _kButtonHeight;
+    }
+
+    final size = constraints.constrain(const Size(double.infinity, double.infinity));
+
+    // Size alert dialog content.
+    final Size maxContentSize = new Size(size.width, size.height - minActionSpace);
+    _content.layout(
+      new BoxConstraints.loose(maxContentSize),
+      parentUsesSize: true,
+    );
+    final Size contentSize = _content.size;
+
+    // Size alert dialog actions.
+    final Size maxActionSize = new Size(
+      size.width,
+      size.height - contentSize.height,
+    );
+    _actions.layout(
+      new BoxConstraints.loose(maxActionSize),
+      parentUsesSize: true,
+    );
+    final Size actionsSize = _actions.size;
+
+    // Calculate overall dialog height.
+    final double dialogHeight = contentSize.height + actionsSize.height;
+
+    this.size = new Size(size.width, dialogHeight);
+
+    // Layout the blur, content, and the actions.
+    _content.layout(
```

Comment:
I don't think you need to do these 2 steps. You're giving back to the child the size that it itself just gave you.

Reply:
yep, you're right.